crates/core/src/types.rs

```diff
@@ -70,6 +73,7 @@ pub struct BundleWithMetadata {
     bundle: Bundle,
     uuid: Uuid,
     transactions: Vec<OpTxEnvelope>,
+    meter_bundle_response: Option<MeterBundleResponse>,
 }
```

Comment:
if a `Bundle` fails `validate_bundle` then there wouldn't be a time to meter bundle, which is why I wrapped `MeterBundleResponse` around an Option

Reply:
This looks good now, but for context, I'm going to have to stop using `BundleWithMetadata` in `node-reth`'s metering crate since I won't have the `meter_bundle_response` yet. If there ends up being another type with the utility functions on it (I'm using `bundle_with_metadata.bundle_hash()`) then I'll use it, otherwise I'll reintroduce a standalone function for it local to that crate.

Reply:
Yes, the most straightforward thing to do is to never convert it from `Bundle` at all. But it'd be nice to have a way to pull in utility functions from `tips-core`. Right now they're all `BundleWithMetadata` methods, so we'd either want to move them to a new type (in that hypothetical third PR we talked about), change them to standalone utility functions instead of methods, or just reimplement them where `BundleWithMetadata` doesn't fit.